### ISSUE TITLE
Error messages for invalid inputs

### ISSUE THREAD

**Issue by alice221927** [einDamian/discrete-choice#83]
According to our requirements: When performing import, export of derivatives or alternatives or open and save project of projects, which contain invalid functions, an error message must appear. The user can then choose to continue the process or cancel it.

**Comment by alice221927** [comment]
is there a way to check the existence of invalid inputs easily? like a function in model and controller, that return either true or false?
@Mikrischu 
@einDamian 
@Kevin2471 
@FliFlaFloo

**Comment by einDamian** [comment]
Currently there is no way of checking if there are any invalid alternatives/derivatives/... easily with a function.
I'm guessing you want to check for any invalidities and notify the user before saving.
If so, I could imagine a function called smth like 'is_valid()' for the ProjectSnapshot, which checks exactly that

**Comment by alice221927** [comment]
@einDamian 
Can you assist me more with this? I don't know how this should be implemented...
And How can we realize notifying the user about invalid derivative or function before importing it??
@Mikrischu 
@FliFlaFloo 
@Kevin-uxpkw 

Is it possible to change the global test cases from the requirements Document?

**Comment by FliFlaFloo** [comment]
@alice221927 I thought we could generally check if there any functions that contain errors (that's easily doable by checking if they have invalid error reports) Then you only need to remember what the last functions were (by the labels) and give the user the error messages for each of the functions. That can really be done by the view (maybe in combination with the controller). You only need a list in one of them with labels of functions where the criteria needs to be checked. I have this task on my todo list anyways, sorry I thought it was only my responsibility, else I would have communicated with you. I have started implementing this already in the way I described. Should be done by tomorrow
